pkg/realserver/configure.go

```diff
@@ -338,6 +379,96 @@ func (r *realserver) periodic() error {
 	}
 }
 
+// ConfigureHAProxy this function is the bridge between a v6 address and a v4
+// pod address. This function iterates over the declared v6 configs and backends
+// for each, checks if any pods match that service selector, and creates a
+// haproxy instance for each backend that maps the VIP:PORT to a list of backend
+// these are the pod ips, not the service IPs, to ensure traffic stays on-node
+func (r *realserver) ConfigureHAProxy() error {
+
+	configSet := []haproxy.VIPConfig{}
+	for ip, config := range r.config.Config6 {
+		// make a single haproxy server for each v6 VIP with all backends
+		for port, service := range config {
+			// fetch the service config and pluck the clusterIP
+			if !r.node.HasServiceRunning(service.Namespace, service.Service, service.PortName) {
+				r.logger.Warnf("no service found for configuration [%s]:(%s/%s), skipping haproxy config", string(ip), service.Namespace, service.Service)
+				continue
+			}
+
+			ips := r.node.GetPodIPs(service.Namespace, service.Service, service.PortName)
+
+			services := r.watcher.Services()
```

Comment:
because the LB port can differ from the service port, and the service port can differ from the pod port, attempting to get the `targetPort` of the service. needs a comment!

Reply:
commented